internal/handler/flight.go

```diff
@@ -2,16 +2,50 @@ package handler
 
 import (
 	"context"
+	"github.com/dilyara4949/flight-booking-api/internal/domain"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/request"
 	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
 	"github.com/gin-gonic/gin"
+	"log/slog"
 	"github.com/google/uuid"
 	"net/http"
 )
 
 type FlightService interface {
+	Create(ctx context.Context, flight request.CreateFlight) (domain.Flight, error)
 	Delete(ctx context.Context, id uuid.UUID) error
 }
 
+func CreateFlightHandler(service FlightService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var req request.CreateFlight
+
+		err := c.ShouldBind(&req)
+		if err != nil {
+			slog.Error("error at binding request body", "error", err.Error())
+
+			c.JSON(http.StatusBadRequest, response.Error{Error: "error at binding request body"})
+
+			return
+		}
+
+		if !ValidateFlightRequest(req) {
```

Comment:
Overall it would be preferrable to have `req.Validate()` and return an `error` specifying validation error(s). The way it handled cutrrently, FE has no information on which fields are missing, which would be useful for marking them for user.